### DISCUSSION TITLE
Block progress based on previous responses to same item

### DISCUSSION THREAD

**Discussion by LaraKroencke** [rubenarslan/formr.org#541] [Q&A]
Hi,

I am currently programming a survey called “Yourlife_Select_exercises” which includes the multiple choice item “challenges” (type: mc_multiple). Participants complete the survey multiple times and each time they do so, they are requested to select at least one new option for the "challenges" item that they haven't selected previously. To ensure this condition is met, I’ve tried using a block item which should block progress if no new options are selected (see JavaScript code below):

Yourlife_Select_exercises.challenges[Yourlife_Select_exercises.challenges.length - 1].split(", ").every(number => Yourlife_Select_exercises.challenges.slice(0, -1).flatMap(vector => vector.split(", ")).includes(number)) //js_only

However, formR lets me progress regardless of the options selected, and no error messages are provided to indicate the source of the problem. I’ve tried multiple variations of this code and also R code instead of JavaScript but nothing seems to work. Any help would be greatly appreciated!

Thank you in advance
Lara

**Comment by rubenarslan** [comment]
To debug Javascript problems, you should open the Developer Console in your browser while running the study.

Off the top of my head, to solve this problem, you need

1. Javascript (because the block item is interactive)
2. The information which options they've previously selected. I don't understand enough about your aim to give you the solution, but it'll probably involve an item of type "hidden" with something like `last(Yourlife_Select_exercises $challenges)` in the value column. Maybe name it something like "last_challenges".
3. Then your Javascript showif for the block item has to compare the new selection (challenges) with the old (last_challenges).

**Reply by LaraKroencke** [reply]
Hi Ruben,

thank you for your response! I managed to create a new variable (last_challenges) which stores the options that participants have previously selected, but I can’t get the JavaScript code which compares the new selection (challenges) with the old (last_challenges) to work.

This is the code I used:
challenges.split(", ").every(number => last_challenges.split(", ").includes(Number(number))) //js_only

I also attached the survey here.

All the best
Lara

[Yourlife_Select_exercises.ods](https://github.com/rubenarslan/formr.org/files/15286602/Yourlife_Select_exercises.ods)

**Reply by rubenarslan** [reply]
I think you just have to lose the `Number()` call (which coerces the last challenges to numbers, but not the new challenges, so it will always yield false).

```javascript
var challenges = "rollerskate_nude, kiss_a_frog, sing_a_song_on_subway, contradict_your_boss"
var last_challenges = "kiss_a_frog, contradict_your_boss"
console.log(challenges.split(", "));
console.log(challenges.split(", ").every(challenge => last_challenges.split(", ").includes(challenge))); //js_only

var challenges = "contradict_your_boss"
var last_challenges = "kiss_a_frog, contradict_your_boss"
console.log(challenges.split(", "));
console.log(challenges.split(", ").every(challenge => last_challenges.split(", ").includes(challenge))); //js_only


var challenges = "1, 2, 4"
var last_challenges = "1, 2, 3"
console.log(challenges.split(", "));
console.log(challenges.split(", ").every(challenge => last_challenges.split(", ").includes(challenge))); //js_only

```

Here's a JS Fiddle to try it out:
https://jsfiddle.net/b0a3y8f7/
